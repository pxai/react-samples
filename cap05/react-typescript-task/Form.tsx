import * as React from 'react';
import Task from "./Task";

interface FormProps {
    task?: Task
    onUpdate: (task) => void
}

interface FormState {
    name: string
}

class Form extends React.Component<FormProps, FormState> {
    public constructor (props: FormProps) {
        super(props);

        this.state =  { name: props.task ? props.task.name : "" };
    }

    public render (): React.ReactNode {
        return <div className="Form">
            <div><label>Task</label></div>
            <div>
                <input type="text" onChange={this.updateTask.bind(this)} value={this.state.name} />
            </div>
            <div>
                <input type="button" value="Save" onClick={this.save.bind(this)} />
            </div>
        </div>;
    }

    private updateTask (event) {
        this.setState({ name: event.target.value });
    }

    private save () {
        if (this.state.name.trim() !== "") {
            this.props.onUpdate({name: this.state.name});
            this.setState({name: ""});
        }
    }
}

export default Form;