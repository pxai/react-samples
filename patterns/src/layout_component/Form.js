import React, { Component } from "react";

class Form extends Component {
    shouldComponentUpdate() {
        return false;
    }

    render() {
        return <form>
            <div>{this.props.input}</div>
            <div>{this.props.button}</div>
        </form>;
    }
}

export default Form;