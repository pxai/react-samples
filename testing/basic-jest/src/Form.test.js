import React from 'react';
import ReactDOM from 'react-dom';
import Form from './Form';
import renderer from 'react-test-renderer';

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Form />, div);
  ReactDOM.unmountComponentAtNode(div);
});


test('it matches snapshot', () => {
	const component = renderer.create(<Form />);
 let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

