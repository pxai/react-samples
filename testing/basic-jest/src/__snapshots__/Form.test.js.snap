// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<div>
  <div>
    <label>
      Task
    </label>
  </div>
  <div>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div>
    <input
      onClick={[Function]}
      type="button"
      value="Save"
    />
  </div>
</div>
`;
